server/system/types/smtp_server.go

```diff
@@ -0,0 +1,11 @@
+package types
+
+type (
+	SmtpServer struct {
```

Comment:
Rename this to something more appropriate. 

_What it "says on a box" is not what it does._ 

Why did you decide to use a struct (and not use the fn. args)?